bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/ConfigDescriptionParameter.java

```diff
@@ -183,28 +192,48 @@ public ConfigDescriptionParameter(String name, Type type, BigDecimal minimum, Bi
         this.max = maximum;
         this.step = stepsize;
         this.pattern = pattern;
-        this.readOnly = readOnly;
-        this.multiple = multiple;
-        this.advanced = advanced;
-
         this.context = context;
-        this.required = required;
         this.defaultValue = defaultValue;
         this.label = label;
         this.description = description;
+        this.multipleLimit = multipleLimit;
 
-        if (options != null)
+        if (readOnly == null) {
+            this.readOnly = false;
```

Comment:
This is imho not fully clean. In the noargs constructor you keep the `this.readonly` uninitialized while you specifically set it here to false. I would prefer to have both constructors operate the same way. The easiest is probably to explicitly initialize the field in the class like `boolean readOnly = false;` and only keep the `!=` case here in the constructor. Same for the other booleans.